packages/create-component/scaffolders/constants.js

```diff
@@ -0,0 +1,29 @@
+const renames = {
```

Comment:
couldn't get this need da

Reply:
We have a 'project template' inside this codebase bro,
![image](https://github.com/user-attachments/assets/b3507696-bc68-4cfd-86a3-d385fd75cf4b)

The scaffolder basically copies this template folder to the user's computer and makes some modifications to it.

So,
![image](https://github.com/user-attachments/assets/3b08f41e-2f38-4f6d-aa02-61ca84cbe8a4)

_gitignore will become .gitignore when the project is created, _npmrc will become .npmrc and so on.

## Why not to store .gitignore as .gitignore but as _gitignore (in the template - subdirectory)?

If we were to store .gitignore in a subdirectory, it might accidentally ignore some folders/files placed inside the template folder, which might break our npm release pipeline... So I am storing it as _gitignore and when it goes to the user's computer it become .gitignore. @rakeshar3796